### PR TITLE
fix: Initial conditional expression support

### PR DESCRIPTION
**Pull request by pleacu**
Fixes: #1860

Successfully parse existing mapping conditional expressions

sc1

No widget tokenization yet - that's next.

### CODE REVIEW

**Comment by pleacu** [comment]
@riccardo-forina - wdyt?  This gets the expression component  classes introduced.

File: ui-react/packages/atlasmap-core/src/react/Atlasmap.tsx [hunk 1]
```diff
@@ -235,9 +239,18 @@ export const Atlasmap: FunctionComponent<IAtlasmapProps> = ({
         onShowMappingPreview={toggleMappingPreview}
         onShowMappedFields={toggleShowMappedFields}
         onShowUnmappedFields={toggleShowUnmappedFields}
+        onConditionalMappingExpressionEnabled={
+          onConditionalMappingExpressionEnabled
+        }
+        onGetMappingExpressionStr={onGetMappingExpressionStr}
+        onToggleExpressionMode={() => {
+          onToggleExpressionMode();
+          expressionTokens = [''];
```
Comment (riccardo-forina) [inline ui-react/packages/atlasmap-core/src/react/Atlasmap.tsx]: Uhm, usually this in the context of a React component is an error. There is no need to "reset" the expressionTokens, if that's the goal. Once the rendering function returns any volatile variable will be lost.

File: ui-react/packages/atlasmap-core/src/services/mapping-management.service.ts [hunk 2]
```diff
@@ -579,6 +579,15 @@ export class MappingManagementService {
     }
   }
 
+  conditionalMappingExpressionEnabled(): boolean {
+    if (!this.cfg || !this.cfg.mappings || !this.cfg.mappings.activeMapping) {
+      return false;
+    };
+    return (this.cfg.mappings && this.cfg.mappings.activeMapping &&
```
Comment (riccardo-forina) [inline ui-react/packages/atlasmap-core/src/services/mapping-management.service.ts]: This return can be simplified to `return !!(this.cfg.mappings?.activeMapping?.transition?.enableExpression?)` with the new TS

File: ui-react/packages/atlasmap-ui/src/AtlasmapUI/AtlasmapCanvasView/AtlasmapCanvasView.tsx [hunk 3]
```diff
@@ -47,12 +55,26 @@ export const AtlasmapCanvasView: FunctionComponent<IAtlasmapCanvasViewProps> = (
...
+    return onConditionalMappingExpressionEnabled();
+  }, [onConditionalMappingExpressionEnabled]);
+
+  const getMappingExpressionStr = useCallback(() => {
+    return onGetMappingExpressionStr();
+  }, [onGetMappingExpressionStr]);
+
+  const toggleExpressionMode = useCallback(() => {
+    onToggleExpressionMode();
+    expressionTokens[0] = 'ratsky';
```
Comment (riccardo-forina) [inline ui-react/packages/atlasmap-ui/src/AtlasmapUI/AtlasmapCanvasView/AtlasmapCanvasView.tsx]: Same comment as above about setting and resetting variables inside event handlers.

File: ui-react/packages/atlasmap-ui/src/CanvasView/components/CanvasViewToolbar.tsx [hunk 4]
```diff
@@ -35,18 +45,23 @@ export const CanvasViewToolbar: FunctionComponent<ICanvasViewToolbarProps> = ({
           <Button
             variant={'plain'}
             aria-label="Enable/ Disable conditional mapping expression"
-            onClick={onToggleExpressionMode}
+            onClick={() => {
+              onToggleExpressionMode();
+              // setCondExprEnabled(onConditionalMappingExpressionEnabled());
+              condExprEnabled = onConditionalMappingExpressionEnabled();
```
Comment (riccardo-forina) [inline ui-react/packages/atlasmap-ui/src/CanvasView/components/CanvasViewToolbar.tsx]: Sorry @pleacu, but this is the same thing as the previous comment. You can't change a local variable in an event handler in React, it makes no sense.

Comment (pleacu) [reply ui-react/packages/atlasmap-ui/src/CanvasView/components/CanvasViewToolbar.tsx]: Yes - will address that today.  My issue was that I was calling onConditionalMappingExpressionEnabled() to set a useState boolean.  The core service function (conditionalMappingExpressionEnabled) returns true but the onConditionalMappingExpressionEnabled() returns false in the onClick():

setCondExprEnabled(onConditionalMappingExpressionEnabled());

Is there a special syntax to invoke onConditionalMappingExpressionEnabled()??

**Pull request merged**